pytket/extensions/qujax/qujax_convert.py

```diff
@@ -37,17 +37,26 @@ def _tk_qubits_to_inds(tk_qubits: Sequence[Qubit]) -> Tuple[int, ...]:
 
 
 def tk_to_qujax_args(
-    circuit: Circuit,
+    circuit: Circuit, symbol_map: Optional[Union[bool, dict]] = False
 ) -> Tuple[Sequence[str], Sequence[Sequence[int]], Sequence[Sequence[int]], int]:
     """
     Converts a pytket circuit into a tuple of arguments representing
     a qujax quantum circuit.
     Assumes all circuit gates can be found in qujax.gates.
-    Parameter argument of created function will be ordered as in circuit.get_commands()
-    - pytket automatically reorders some gates, consider using Barriers.
+    If symbol_map is False, parameter arguments of created function are ordered as in
```

Comment:
Possibly we need to elaborate on this API?

Particularly to make clear the difference between `symbol_map=False` and `symbol_map=True`

Reply:
For me it is not clear yet what the behaviour of this function will be if there are any symbolic parameter in the circuit and the  symbol_map is False. What is the behaviour in that case?

Reply:
If symbol_map is False, the function will ignore whether the parameter is symbolic or not, ie for each gate it just looks at how many parameters there is and assume that all the parameters are independent

Reply:
I tried to improve the description; I use "qujax circuit parameters" here which seems to be in agreement with https://cqcl.github.io/qujax/api/get_params_to_statetensor_func.html -- let me know what you think

Reply:
Maybe the below is clearer?

>     If symbol_map is False, any symbolic tket parameters will be ignored. Parameterised gates will be determined
>     based on whether they are stored as functions (parameterised) or arrays (unparameterised) in qujax.gates. The order
>     of parameters is the same as in circuit.get_commands().

Reply:
I think it might be helpful to separate this two questions in two parameters. For me it is still not 100 % clear. Maybe we could improve on this questions with having a `symbolic_conversion` and `symbolic_map` (with default value None?)

Reply:
> Maybe the below is clearer?
> 
> > ```
> > If symbol_map is False, any symbolic tket parameters will be ignored. Parameterised gates will be determined
> > based on whether they are stored as functions (parameterised) or arrays (unparameterised) in qujax.gates. The order
> > of parameters is the same as in circuit.get_commands().
> > ```

Symbolic parameters are not really ignored, they are just treated in the same way as numerical ones, right? In practice, structure of parameters is also just taken from tket and we don't even check that it is the same in qujax gates?

Reply:
> I think it might be helpful to separate this two questions in two parameters. For me it is still not 100 % clear. Maybe we could improve on this questions with having a `symbolic_conversion` and `symbolic_map` (with default value None?)

This would be a better idea, if we just let symbolic_conversion = False ignore what symbolic_map is, even if it is not None

Reply:
> I think it might be helpful to separate this two questions in two parameters. For me it is still not 100 % clear. Maybe we could improve on this questions with having a `symbolic_conversion` and `symbolic_map` (with default value None?)

I am not convinced this is more clear, it might lead to confusion in the case that symbolic_map is given but symbolic_conversion is not. And we would anyway need to properly document what happens when symbolic_conversion=True but symbolic_map=None (plus the other combinations)

Reply:
I would just given an error if symbolic_conversion=False and symbolic_map is not None (to make sure this is understood from the user and no unexpected behaviour is created?

Reply:
(Only making suggestions, if you are both happy with something I will not stop you from doing that)

Reply:
One more idea I had would be sticking with one parameter with the options, None, empty map and not empty map. Maybe that is more clear and easy to explain? (with the default value None)

Reply:
@cqc-melf thank you very much for your helpful comments!

I think my preference is to stay with the single `symbol_map: Union[bool, dict]` argument.
I think with clear documentation this is the cleanest API